### PR TITLE
[Merged by Bors] - Improve block-lookup functionality

### PR DESCRIPTION
**Pull request by AgeManning**
Improves some of the functionality around single and parent block lookup. 

Gives extra information about whether failures for lookups are related to processing or downloading.

This is entirely untested.

### CODE REVIEW

File: beacon_node/network/src/sync/block_lookups/single_block_lookup.rs [hunk 1]
```diff
@@ -197,13 +218,4 @@ mod tests {
         sl.request_block().unwrap();
         sl.verify_block(Some(Box::new(block))).unwrap().unwrap();
     }
-
-    #[test]
-    fn test_max_attempts() {
```
Comment (AgeManning) [inline beacon_node/network/src/sync/block_lookups/single_block_lookup.rs:203]: This test looked like it did nothing, so i removed it

File: beacon_node/network/src/sync/block_lookups/mod.rs [hunk 2]
```diff
@@ -256,7 +261,6 @@ impl<T: BeaconChainTypes> BlockLookups<T> {
                     self.request_parent(parent_lookup, cx);
                 }
                 VerifyError::PreviousFailure { parent_root } => {
-                    self.failed_chains.insert(parent_lookup.chain_hash());
```
Comment (AgeManning) [inline beacon_node/network/src/sync/block_lookups/mod.rs:261]: This was a duplicate, so removed it

**Review by divagant-martian — APPROVED** [review]
Thanks for the added docs :)

**Comment by divagant-martian** [comment]
I'm adding tests to this, so please refrain from merging

**Comment by divagant-martian** [comment]
@AgeManning give https://github.com/sigp/lighthouse/pull/3287/commits/5ec346fa83b7d03c5c7f942475895544e5d5209b a look

**Comment by AgeManning** [comment]
Awesome thanks @divagant-martian. Nice catch, i've left some comments on the commit.

**Comment by divagant-martian** [comment]
@AgeManning fresh out of the oven https://github.com/sigp/lighthouse/pull/3287/commits/ac29de369bc7450782db00ce3d84149c31a3f82d

**Comment by AgeManning** [comment]
Nice, lets dooo it.

**Pull request closed without merging**